x-pack/plugins/observability_solution/investigate_app/server/services/get_alerts_client.ts

```diff
@@ -19,12 +26,11 @@ export async function getAlertsClient({
   const ruleRegistryPluginStart = await plugins.ruleRegistry.start();
   const alertsClient = await ruleRegistryPluginStart.getRacClientWithRequest(request);
   const alertsIndices = await alertsClient.getAuthorizedAlertsIndices([
-    'logs',
-    'infrastructure',
-    'apm',
-    'slo',
-    'uptime',
-    'observability',
```

Comment:
Does the `observability` match to more rule type IDs I have put? Should I add all o11y rule type IDs?

Reply:
I think here we need to get all alert indices (where user has access) in Observability to find related alerts, so it would be good to use `OBSERVABILITY_RULE_TYPE_IDS`. 

This doesn't include ES Query rule though and I am not sure with `OBSERVABILITY_RULE_TYPE_IDS_WITH_STACK` if we can limit the scope so that only ES query alerts created in Observability are returned. This may not be possible right now since ES query alerts are indexed in common `stack*` alert index. @cnasikas please correct me if I am wrong and let me know if you have any suggestion.

Reply:
is it possible to allow `getAuthorizedAlertsIndices` to accept consumers in addition to ruleTypeIds?

Reply:
The indices are not connected to the consumers of the rule. They are connected only to the rule type. You can check how we get the indices from the rule types here https://github.com/elastic/kibana/blob/da85efe5093c148d4b91bcd3e21fd93c9f182a4f/x-pack/plugins/alerting/server/lib/create_get_alert_indices_alias.ts#L14. As you correctly said, ES query alerts are indexed in the `stack*` alert index and you may not want to get these indices here. I changed it to `OBSERVABILITY_RULE_TYPE_IDS` in [`00b50fd` (#183756)](https://github.com/elastic/kibana/pull/183756/commits/00b50fd3339e07bc4593f20bc3ba5a206436a495). Let me know if you need to change it to `OBSERVABILITY_RULE_TYPE_IDS_WITH_STACK`.

Reply:
Thanks @cnasikas for the change. I think using `OBSERVABILITY_RULE_TYPE_IDS` works for now. In case we use `OBSERVABILITY_RULE_TYPE_IDS_WITH_STACK` to get alert indices, I need to filter alerts with consumer so that only observability alerts are returned for ES query rule type. I will open a PR for that change after this PR is merged.